src/vm/tieredcompilation.cpp

```diff
@@ -285,6 +337,19 @@ void TieredCompilationManager::AsyncPromoteMethodToTier1(MethodDesc* pMethodDesc
         }
     }
 
+    if (m_methodsPendingCountingForTier1 != nullptr)
```

Comment:
I assume the intent of this check is along the lines of 

    if(IsDelayActive())

If so it might be useful to make a tiny inlinable wrapper and use that. At some point when we better understand circumstances when the delay is useful we might want it to activate for conditions that don't have any methods pending call counting.

Reply:
Yep will do